fbs/rdg.fbs

```diff
@@ -0,0 +1,145 @@
+namespace katana.fbs;
+
+enum RDGAccessType : byte {
+  All,
+  Topology,
+  AllProperties,
+  NamedNodeProperty,
+  NamedEdgeProperty,
+}
+
+enum RDGAccessMode : byte {
+  Write,
+  Read,
+}
+
+// An RDGAccess is a reference to an abstract location for the purpose of
+// conflict detection.
+table RDGAccess {
+  rdg_name : string;
+
+  // Write accesses set the versions zero to indicate the latest (uncommitted)
+  // version. Only historical or snapshot reads will have non-zero versions.
+  rdg_version : uint64;
+  metadata_version : uint64;
+
+  mode : RDGAccessMode;
+
+  type : RDGAccessType;
+  // property_name is empty unless type is NamedNodeProperty or
+  // NamedEdgeProperty.
+  property_name : string;
+}
```

Comment:
I don't like the name "access" here. What about "RDGAccessEvent" or something like that. I initially assumed these were about controlling access or permissions like ACLs.

Reply:
If you think "access" is confusing, I don't think "access event" is any better.

Originally, these were called "locations" but no one liked that name either: https://github.com/KatanaGraph/katana-enterprise/pull/2142#discussion_r723373632